lib/rubygems.rb

```diff
@@ -1242,6 +1242,23 @@ class << self
 
   end
 
+  ##
+  # The SOURCE_DATE_EPOCH environment variable (or, if that's not set, the current time), converted to Time object.
+  # This is used throughout RubyGems for enabling reproducible builds.
+  #
+  # If it is not set as an environment variable already, this also sets it.
+  #
+  # Details on SOURCE_DATE_EPOCH:
+  # https://reproducible-builds.org/specs/source-date-epoch/
+
+  def self.source_date_epoch
+    if ENV["SOURCE_DATE_EPOCH"].to_s.empty?
+      ENV["SOURCE_DATE_EPOCH"] = Time.now.to_i.to_s
+    end
+
+    Time.at(ENV["SOURCE_DATE_EPOCH"].to_i).utc.freeze
+  end
+
```

Comment:
What you think of:
```ruby
def self.source_date_epoch
  source_date_epoch = ENV["SOURCE_DATE_EPOCH"]
  source_date_epoch = Time.now.to_i if source_date_epoch.strip.empty?
 
  Time.at(source_date_epoch).utc.freeze
end
```

Reply:
The entire purpose of this PR is to set `ENV["SOURCE_DATE_EPOCH"]` if it hasn't been set, which your variant doesn't do.